tests/io/FileFactoryTest.py

```diff
@@ -0,0 +1,59 @@
+import os
+import unittest
+
+from coalib.io.FileFactory import FileFactory
+
+
+def get_path_components(filepath):
+    """
+    Splits the filepath into components to provide a unique
+    test results that passes on all CIs.
+    """
+    if filepath[0] == '/' or filepath[0] == '\\':
+        filepath = filepath[1:]
+    components = []
+    while filepath != '':
+        head, tail = os.path.split(filepath)
+        components.append(tail)
+        filepath = head
+    return components
```

Comment:
Actually sorry for maybe not telling earlier, but I've googled a bit around, and there's actually this possibility:
```python
os.path.normpath(path).split(os.sep)
```